torch/_inductor/triton_heuristics.py

```diff
@@ -398,11 +405,22 @@ def run(self, *args, grid, stream):
             launcher.config.pre_hook(
                 {**zip(self.arg_names, args), **launcher.config.kwargs}
             )
-        return launcher(
-            *args,
-            grid=grid,
-            stream=stream,
-        )
+
+        # guard the record_function_ctx and only call it if profiling is currently
+        # in progress, to reduce latency when profiler is not turned on.
```

Comment:
nit: I would also mention that we don't refactor this code to use a noopcontext as this adds too much overhead to this perf-critical part of the code.
I could very well see someone removing that optimization by mistake otherwise